app/src/main/java/com/keylesspalace/tusky/components/conversation/ConversationEntity.kt

```diff
@@ -100,7 +97,7 @@ data class ConversationStatusEntity(
         if (inReplyToId != other.inReplyToId) return false
         if (inReplyToAccountId != other.inReplyToAccountId) return false
         if (account != other.account) return false
-        if (content.toString() != other.content.toString()) return false // TODO find a better method to compare two spanned strings
+        if (content.toString() != other.content.toString()) return false
```

Comment:
Here and in `hashCode()`:
allocating is eh. It is actually `SpannableStringBuilder` as far as I understand and it *will* allocate new array *and* new string.
Probably the best way to do this would be to implement a custom function/class on top of `charAt()` that would compare or hashCode.
Alternatively we could delay parsing HTML until `StatusViewData` which also kinda makes sense to me.

It can be a separate task but I think this would also improve performance quite a bit.

Reply:
I'll look into it

Reply:
Hmmm, that's quite interesting, good call to actually test it. I wonder if it uses cached `hashCode()` on the instance for quick equality check.

Reply:
It might be because of dynamic dispatch on `Spanned` when you call `get()`.

Reply:
Yeah my test is also not super accurate probably. I'll figure it out when i have more time.

Reply:
It was quite a moonshot anyway, I wish they just implemented it properly, then they could do it directly their buffer without dispatch and additional checks but we live in society.